### PR TITLE
PWA capabilities

### PR DESCRIPTION
**Pull request by srivats22**
This PR is fro issue #88 

Added PWA capabilities new dependency and updates main.tsx and vite.config.ts for the same

# Install Popup
Screenshot 2024-11-01 at 7 59 37 AM

# Running as a PWA
Screenshot 2024-11-01 at 7 59 51 AM

# Code also works
Screenshot 2024-11-01 at 8 00 45 AM

# Manifest being read
Screenshot 2024-11-01 at 8 06 31 AM

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -21,6 +21,17 @@ import { ConfigWrapper } from "./Config.tsx";
 import "./index.scss";
 import { AppStateWrapper } from "./AppState.tsx";
 
+if('serviceWorker' in navigator){
```
Comment (romannurik) [inline src/main.tsx:24]: nit: align formatting to rest of the file

File: src/main.tsx [hunk 2]
```diff
@@ -21,6 +21,17 @@ import { ConfigWrapper } from "./Config.tsx";
...
+  navigator.serviceWorker.register('/sw.js', {
+    scope: '/'
+  }).then((register) => {
+    console.log('service worker registered', register)
```
Comment (romannurik) [inline src/main.tsx:28]: nit: don't console.log

File: src/main.tsx [hunk 3]
```diff
@@ -21,6 +21,17 @@ import { ConfigWrapper } from "./Config.tsx";
 import "./index.scss";
 import { AppStateWrapper } from "./AppState.tsx";
 
+if('serviceWorker' in navigator){
+  navigator.serviceWorker.register('/sw.js', {
+    scope: '/'
+  }).then((register) => {
```
Comment (romannurik) [inline src/main.tsx:27]: nit: async/await in a try/catch block

File: vite.config.ts [hunk 4]
```diff
@@ -1,8 +1,54 @@
-import { defineConfig } from 'vite'
-import react from '@vitejs/plugin-react'
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react";
+import { VitePWA, VitePWAOptions } from "vite-plugin-pwa";
+
+const manifestForPlugin: Partial<VitePWAOptions> = {
+  registerType: "autoUpdate",
+  includeAssets: ["favicon.ico"],
+  manifest: {
```
Comment (romannurik) [inline vite.config.ts:8]: can we pull this from the existing manifest.json file using an import?

**Review by romannurik — CHANGES_REQUESTED** [review]
Huge thanks! Dropped some feedback, but this looks nearly ready. If you don't have time feel free to hand off to me to finish.

**Comment by srivats22** [comment]
> Huge thanks! Dropped some feedback, but this looks nearly ready. If you don't have time feel free to hand off to me to finish.

Will work on it and if I am not able to figure it out by this weekend will reachout/re-assing

File: src/main.tsx (hunk 1 above)
Comment (srivats22) [reply src/main.tsx]: Addressed in recent PR

File: src/main.tsx (hunk 2 above)
Comment (srivats22) [reply src/main.tsx]: Addressed in recent PR

File: src/main.tsx (hunk 3 above)
Comment (srivats22) [reply src/main.tsx]: Addressed in recent PR

File: vite.config.ts (hunk 4 above)
Comment (srivats22) [reply vite.config.ts:8]: Tried pulling it via import but didn't work...

**Comment by romannurik** [comment]
Thank you! There were still some issues with this PR so I'm closing it, but I'll submit your changes along with my fixes momentarily! Thanks for getting us 90% of the way there!

**Comment by romannurik** [comment]
Fixes in commit 5871dff66d46888d53eaaed29fafcaf67501ff60

**Pull request closed without merging**